tests/Unit/Evolution/Systems/Burgers/FiniteDifference/Test_BoundaryGhostData.cpp

```diff
@@ -0,0 +1,263 @@
+      typeid(Burgers::BoundaryConditions::DirichletAnalytic)) {
+    const size_t ghost_zone_size{ReconstructorForTest{}.ghost_zone_size()};
+
+    // For the element used in this test, external boundary is located at
+    // the direction of upper_xi
+    const auto direction = Direction<1>::upper_xi();
+    DirectionMap<1, bool> direction_to_slice{};
+    direction_to_slice[direction] = true;
+
+    // Slice subcell logical coordinates
+    auto sliced_logical_coords_v =
+        evolution::dg::subcell::detail::slice_data_impl(
+            gsl::make_span(get<0>(subcell_logical_coords).data(),
+                           get<0>(subcell_logical_coords).size()),
+            subcell_mesh.extents(), ghost_zone_size,
+            direction_to_slice)[direction];
+
+    // Since dg::subcell::detail::slice_data_impl() returns its result in the
+    // form of std::vector<double>, create a Variables object to properly handle
+    // it.
+    Variables<tmpl::list<domain::Tags::Coordinates<1, Frame::ElementLogical>>>
+        temp_vars_to_store_sliced_coords{};
+    temp_vars_to_store_sliced_coords.initialize(sliced_logical_coords_v.size(),
+                                                0.0);
+    std::copy(sliced_logical_coords_v.begin(), sliced_logical_coords_v.end(),
+              temp_vars_to_store_sliced_coords.data());
+
+    // Shift the sliced logical coordinates and compute inertial coordinates.
+    const double delta_xi{get<0>(subcell_logical_coords)[1] -
+                          get<0>(subcell_logical_coords)[0]};
+    auto& sliced_logical_coords =
+        get<domain::Tags::Coordinates<1, Frame::ElementLogical>>(
+            temp_vars_to_store_sliced_coords);
+    get<0>(sliced_logical_coords) =
+        get<0>(sliced_logical_coords) +
+        direction.sign() * delta_xi * ghost_zone_size;
+    const auto shifted_inertial_coords = (*grid_to_inertial_map)(
+        logical_to_grid_map(sliced_logical_coords), time, functions_of_time);
+
+    // Retrieve expected values of U from analytic solution, and pack the
+    // returned DataVector into std::vector<double> for comparison
+    const Scalar<DataVector> expected_ghost_u_dv =
+        get<Burgers::Tags::U>(solution.variables(
+            shifted_inertial_coords, time, tmpl::list<Burgers::Tags::U>{}));
+    const size_t num_ghost_pts{
+        ghost_zone_size *
+        subcell_mesh.extents().slice_away(direction.dimension()).product()};
+    Variables<tmpl::list<Burgers::Tags::U>> var_expected_ghost_u{num_ghost_pts};
+    get<Burgers::Tags::U>(var_expected_ghost_u) = expected_ghost_u_dv;
+    std::vector<double> expected_ghost_u{
+        var_expected_ghost_u.data(),
+        var_expected_ghost_u.data() + var_expected_ghost_u.size()};
+
+    CHECK_ITERABLE_APPROX(expected_ghost_u, fd_ghost_data);
+  }
+  if (typeid(BoundaryConditionType) ==
+      typeid(Burgers::BoundaryConditions::Outflow)) {
+    // Since U = 1.0 on the subcell mesh outflow condition will not throw any
```

Comment:
I wonder if we should try with more rigorous test on this, e.g. set volume mesh velocity to be a large value and check if the outflow condition throws an error.